synapse/storage/schema/__init__.py

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-SCHEMA_VERSION = 77  # remember to update the list below when updating
+SCHEMA_VERSION = 78  # remember to update the list below when updating
```

Comment:
The notes from https://github.com/matrix-org/synapse/pull/15691 are very helpful when trying to understand if this PR is the correct next step in the migration process :bulb: 


And a breakdown of the PRs for this migration process so far:

 - [#15458](https://github.com/matrix-org/synapse/pull/15458/) (N + 1): Bump `SCHEMA_VERSION` to `76`. Add `full_user_id` column, start writing to both the old and new column moving forward
 - [#15537](https://github.com/matrix-org/synapse/pull/15537) (N + 2): Bump `SCHEMA_VERSION` to `77` and `SCHEMA_COMPAT_VERSION` to `76` because we added a `NOT VALID` constraint that only ensures that new inserts/updates always fill in `full_user_id` which isn't backwards compatible because older Synapse versions don't write to the new `full_user_id` column. Also add a background update to backfill `full_user_id` for all rows
 - [#15649](https://github.com/matrix-org/synapse/pull/15649) (N + 3): Bump `SCHEMA_VERSION` to `78`, add a foreground update to finish off the backfill and turn the `NOT VALID` constraint into a valid one -> `full_user_id NOT NULL`, and start reading from `full_user_id`
 - Future PRs needed for `N + 4` and `N + 5` where we stop writing to `user_id` and drop the column

Reply:
Looks good based on the notes linked above but I've never done this kind of thing before and I'm definitely not an authoritative source of knowledge on this.